src/main/java/com/telefonica/iot/cygnus/sinks/OrionSink.java

```diff
@@ -245,13 +229,74 @@ public Status process() throws EventDeliveryException {
                     } else {
                         LOGGER.warn(e.getMessage());
                     } // if else if
+                    
+                    LOGGER.info("Finishing transaction (" + MDC.get(Constants.HEADER_TRANSACTION_ID) + ")");
+                    txn.commit();
+                    txn.close();
+                    return Status.READY;
                 } // if else
-            } finally {
-                LOGGER.info("Finishing transaction (" + MDC.get(Constants.HEADER_TRANSACTION_ID) + ")");
-            } // try catch finally
+            } // try catch
         } else {
+            // get and process as many events as the batch size
+            int currentIndex;
+            
+            for (currentIndex = accumulator.getAccIndex(); currentIndex < batchSize; currentIndex++) {
```

Comment:
So you check the accumulation timeout each time a new event arrives to the buffer, right?